packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -464,15 +472,16 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
             "OptimismPortal: output proposal has not been finalized yet"
         );
 
+        GameType gameType = disputeGameProxy.gameType();
         // The game type of the dispute game must be the respected game type. This was also checked in
         // `proveWithdrawalTransaction`, but we check it again in case the respected game type has changed since
         // the withdrawal was proven.
-        require(disputeGameProxy.gameType().raw() == respectedGameType.raw(), "OptimismPortal: invalid game type");
+        require(respectedGameType[gameType], "OptimismPortal: invalid game type");
 
         // The game must have been created after `respectedGameTypeUpdatedAt`. This is to prevent users from creating
         // invalid disputes against a deployed game type while the off-chain challenge agents are not watching.
         require(
-            createdAt >= respectedGameTypeUpdatedAt,
+            createdAt >= respectedGameTypeUpdatedAt[gameType],
```

Comment:
We may need to approach this slightly differently. A game type can be respected but we should probably not allow people to continue to play the game or we would have to run a challenger for both the old and the new client code, which doesn't really make any sense. We can respect a game type that was created in the past while no longer respecting new games that were created after the fork.

Reply:
Basically, we need to be able to mark a game type as still respected until time T, after which it is no longer respected. This allows the challenger to stop responding to new games in that game type safely.

Reply:
Updated to use a rolling window here